### PR TITLE
Feature/queue build

### PR DESCRIPTION
**Pull request by fatal10110**

### CODE REVIEW

File: azuredevops/builds.go [hunk 1]
```diff
@@ -47,3 +47,35 @@ func (s *BuildsService) List(opts *BuildsListOptions) ([]Build, error) {
 
 	return response.Builds, err
 }
+
+// BuildsListOptions describes what the request to the API should look like
+type QueueBuildOptions struct {
```
Comment (benmatselby) [inline azuredevops/builds.go]:
```suggestion
// QueueBuildOptions describes what the request to the API should look like
type QueueBuildOptions struct {
```

File: azuredevops/builds.go [hunk 2]
```diff
@@ -47,3 +47,35 @@ func (s *BuildsService) List(opts *BuildsListOptions) ([]Build, error) {
...
+	IgnoreWarnings bool   `url:"ignoreWarnings,omitempty"`
+	CheckInTicket  string `url:"checkInTicket,omitempty"`
+}
+
+type QueueBuildResponse struct {
```
Comment (benmatselby) [inline azuredevops/builds.go]: Could you please add a description here

File: azuredevops/azuredevops.go [hunk 3]
```diff
@@ -77,7 +78,15 @@ func (c *Client) NewRequest(method, URL string, body interface{}) (*http.Request
 // NewBaseRequest does not take into consideration the project
 // and simply uses the base https://%s.visualstudio.com base URL
 func (c *Client) NewBaseRequest(method, URL string, body interface{}) (*http.Request, error) {
-	var buf io.ReadWriter
+	var buf io.Reader
```
Comment (benmatselby) [inline azuredevops/azuredevops.go]: Could you provide some reasoning behind this please?

Comment (fatal10110) [reply azuredevops/azuredevops.go]: Thank you for review! 
`NewRequest` ignores `Writer` interface, it takes only `Reader`, so you don't need the `ReadWriter` actually.

https://golang.org/pkg/net/http/#NewRequest

**Pull request closed without merging**